guides/templates_lambda.md

```diff
@@ -0,0 +1,276 @@
+
+### lib/solar-system.ts
+
+You will need to import the following constructs:
+
+    import { SolarSystemExtensionStack, SolarSystemExtensionStackProps } from '@cdk-cosmos/core';
+    import { SsmState } from '@cosmos-building-blocks/common';
+    import { Function, Runtime, Code } from '@aws-cdk/aws-lambda';
+    import { AppGalaxyStack } from '.';
+
+
+Add tag to solar system props
+
+    export interface AppSolarSystemProps extends SolarSystemExtensionStackProps {
+      appVersion?: string;
+    }
+
+Add readonly bucket inside export statement
+
+    readonly bucket: Bucket;
+
+Add the following inside the constructor, after `super();`. Here, you are using the codeBucket from `lib/cosmos.ts` and the shared vpc through the portal construct. version state is managed through a CDK Cosmos feature.
+
+    const { appVersion } = props || {};
+    const { codeBucket } = this.galaxy.cosmos;
+    const { vpc } = this.portal;
+    
+    const versionState = new SsmState(this, 'VersionState', {
+      name: '/' + this.nodeId('VersionState', '/'),
+      value: appVersion,
+    });
+
+    //handler must be the same as the export from your app
+    const serverFunction = new Function(this, 'ServerFunction', {
+      handler: 'app.lambdaHandler',
+      code: Code.fromBucket(codeBucket, `function-${appVersion}.zip`),
+      runtime: Runtime.NODEJS_12_X,
+      environment: {
+        REGION: 'ap-southeast-2',
+      },
+      vpc,
+    });
+
+> Note: This template uses the SsmState construct, which has an active issue in the Cosmos backlog - (Issue #318)[https://github.com/cdk-cosmos/cosmos/issues/318]. Currently, you will need to pass a hardcoded appVersion on the first run of the pipeline so that a parameter is added to the AWS Parameter Store in each solar system. See details below on setting up the sample lambda app.
+
+    const versionState = new SsmState(this, 'VersionState', {
+      name: '/' + this.nodeId('VersionState', '/'),
+      value: 1,
+    });
+
+#### To add front end you will need a few extra resources:
+imports
+
+    import { RecordTarget, ARecord } from '@aws-cdk/aws-route53';
+    import { LoadBalancerTarget } from '@aws-cdk/aws-route53-targets';
+    import { ApplicationTargetGroup, ListenerCondition } from '@aws-cdk/aws-elasticloadbalancingv2';
+    import { LambdaTarget } from '@aws-cdk/aws-elasticloadbalancingv2-targets';
+
+in constructor
+
+    const { alb, httpsListener } = this.portal.addEcs();
+    const recordName = `slackbot.${this.portal.zone.zoneName}`;
+
+    const serverTargetGroup = new ApplicationTargetGroup(this, 'ServerTargetGroup', {
+      targets: [new LambdaTarget(serverFunction)],
+    });
+
+    httpsListener.addTargetGroups('ServerTargets', {
+      conditions: [ListenerCondition.hostHeaders([recordName])],
+      priority: 9500,
```

Comment:
dunno what the priority should be - this number is arbitrary - there's actually a cosmos function that figures out the priority for you - we can change this later though - 9500 will do for now